### PR TITLE
feat(aapd-837): add dynamic routing and tables with test data

### PR DESCRIPTION
**Pull request by Maby0**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-

## Description of change

Add placeholder index files for all routes with placeholder data in the tables

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/web-comment/src/routes/index.js [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
+ * @returns {string[]}
  */
-const routes = dirNames.reduce((acc, dirName) => {
-	const { router } = require(`./${dirName}`);
-	return {
-		...acc,
-		[`/${dirName}`]: router
-	};
-}, {});
+function getRoutes(directory = __dirname, includeRoot = false) {
```
[the remark this answers is not in the document]
Comment (Maby0) [reply packages/web-comment/src/routes/index.js:12]: I agree that'd be a good shout. It's going to be a bit finicky to mock the right things so I might leave it until later when I've finished the main features of the interested party tickets.

File: packages/appeals-service-api/src/db/seed/data-dev.js [hunk 2]
```diff
@@ -61,22 +118,167 @@ const appeals = [
...
+		appealTypeCode: 'HAS',
+		appealTypeName: 'Householder',
+		decision: '',
+		originalCaseDecisionDate: pickRandom(datesNMonthsAgo(1)),
+		costsAppliedForIndicator: false,
+		LPAApplicationReference: '12/2323233/PLA',
+		siteAddressLine1: '123 Fake Street',
+		siteAddressTown: 'Testville',
+		siteAddressCounty: 'Countyshire',
+		siteAddressPostcode: 'BS1 6PN',
```
Comment (beejjacobs) [inline packages/appeals-service-api/src/db/seed/data-dev.js:175]: Shall we pull out these common bits so we can spread them in?

File: packages/web-comment/src/routes/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (beejjacobs) [reply packages/web-comment/src/routes/index.js:12]: We're also going to pull this into `common` at some point.

File: packages/web-comment/src/utils/appealSorting.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const getOpenAppeals = (appeals) => {
+	return appeals.filter((appeal) => new Date(appeal.interestedPartyRepsDueDate) > currentDate);
+};
+
+const getClosedAppeals = (appeals) => {
+	return appeals.filter((appeal) => new Date(appeal.interestedPartyRepsDueDate) < currentDate);
+};
+
+// should we use this type of function declaration?
+Array.prototype.sortByInterestedPartyRepsDueDate = function () {
```
Comment (beejjacobs) [inline packages/web-comment/src/utils/appealSorting.js:11]: Generally bad practice to alter prototypes - a reason this isn't just a normal sort function? Something like:

```
function sortByInterestedPartyRepsDueDate(a, b) {
  return new Date(a.interestedPartyRepsDueDate) - new Date(b.interestedPartyRepsDueDate);
}

myArr.sort(sortByInterestedPartyRepsDueDate);`
```

File: packages/web-comment/src/utils/formatAddress.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+	formatAddress: (appeal) => {
+		const addressComponents = [
+			appeal.siteAddressLine1,
+			appeal.siteAddressLine2,
+			appeal.siteAddressTown,
+			appeal.siteAddressPostcode
+		];
+
+		return addressComponents.filter((component) => !!component).join(', ');
```
Comment (beejjacobs) [inline packages/web-comment/src/utils/formatAddress.js:10]: I prefer `filter(Boolean)` but either is fine!

File: packages/web-comment/src/utils/getPostcodeSearchResults.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+const fetch = require('node-fetch');
+const { appealsApi } = require('../server.config');
+
+module.exports = {
```
Comment (beejjacobs) [inline packages/web-comment/src/utils/getPostcodeSearchResults.js:4]: I think a better approach is an `AppealsApiClient` class, which can take the baseUrl as a constructor parameter, then have methods for specific endpoints. Thoughts?

File: packages/web-comment/src/routes/comment-appeal/appeals/index.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+	postcodeSearchResults.forEach((appeal) => (appeal.formattedAddress = formatAddress(appeal)));
+
+	const openAppeals = getOpenAppeals(postcodeSearchResults);
+	openAppeals.sortByInterestedPartyRepsDueDate(openAppeals);
+
+	const closedAppeals = getClosedAppeals(postcodeSearchResults);
+	closedAppeals.sortByCaseReference();
+
+	res.render(`comment-appeal/appeals/index`, { postcode, openAppeals, closedAppeals });
```
Comment (beejjacobs) [inline packages/web-comment/src/routes/comment-appeal/appeals/index.js:27]: Be good to use a pattern we have elsewhere for this. Have this in a `controller.js` file, then import into the router and wrap with `asyncHandler` - that way if any promises reject, the error gets handled.

e.g. see https://github.com/Planning-Inspectorate/appeal-planning-decision/commit/2b22d191344252d276f454b341266464306d71dc

**Review by beejjacobs — COMMENTED** [review]
Good amount of seed data, and generally this looks good to me!

A few style/pattern questions to look through for you to consider.

File: packages/appeals-service-api/src/db/seed/data-dev.js (hunk 2 above)
Comment (Maby0) [reply packages/appeals-service-api/src/db/seed/data-dev.js]: done!

File: packages/web-comment/src/utils/appealSorting.js (hunk 3 above)
Comment (Maby0) [reply packages/web-comment/src/utils/appealSorting.js]: agreed. Have changed this.

File: packages/web-comment/src/utils/formatAddress.js (hunk 4 above)
Comment (Maby0) [reply packages/web-comment/src/utils/formatAddress.js]: hadn't seen this before! Changed it to this.

File: packages/web-comment/src/utils/getPostcodeSearchResults.js (hunk 5 above)
Comment (Maby0) [reply packages/web-comment/src/utils/getPostcodeSearchResults.js]: Yeah that's definitely a cleaner solution. Implemented that

File: packages/web-comment/src/utils/appealSorting.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+const currentDate = new Date();
```
[the remark this answers is not in the document]
Comment (Maby0) [reply packages/web-comment/src/utils/appealSorting.js]: nice spot, have instantiated it in each function.

File: packages/web-comment/src/routes/comment-appeal/appeals/index.js (hunk 6 above)
Comment (beejjacobs) [reply packages/web-comment/src/routes/comment-appeal/appeals/index.js]: I'd intended just a controller file in each route folder, I will refactor!

**Review by topickering — APPROVED** [review]
LGTM!

**Review by topickering — APPROVED** [review]

**Review by topickering — APPROVED** [review]

**Pull request merged**